dashboard/app/views/levels/editors/fields/_default_song.html.haml

```diff
@@ -2,5 +2,6 @@
   Default Song
 
 #defaultSong.in.collapse
+  %p Please note - If you are limiting songs in the Limit Song Selection input, the Default Song must be one of those songs.
```

Comment:
There is a limitation here in that the user could still choose a default song not in the list.

Better UX would be to adjust the default song on the fly based upon changing song selection.

Otherwise, if we'd attempted to restrict this down to only the songs actually in the song selection list, the user would need to reload the page in order to select a new song, if something was newly added. I felt it was easier/quicker to just give the warning to make sure they manually match.

I can be persuaded, but would ask for some help with the ruby to get it wired up properly if we feel limiting/updating would be necessary.

Reply:
What happens if a levelbuilder selects a default song not in the limited song list?

Reply:
It basically just ignores it. _Technically_ it's going to try to select it in the drop down, but when it's unfound it'll just default to whatever the first song is in the list.

So the effect on the user is that the first song in the list is selected; event on the content writer is possible confusion as to why their chosen default song isn't populated and selected in the list.

Reply:
Thanks! I suggest swapping the order of the Default Song selector and the Limit Song selection on the edit page so that content editors first limit songs (if they choose to), then select default song.

Reply:
And thanks for the updated screenshot!

Reply:
Would you mind also displaying the current default song?

Reply:
Suggested from a person with a terrible short memory. 😁

Reply:
I'm also thinking about the case when a content editor selects a default song that is not in the limited song list. It looks like that default song is saved and remains checked next time the edit page is open. What do you think about adding logic so that if the default song is not in the limited song list, then the default song is saved as the first song in the limited song list and then is displayed as such  on the edit page next time?

Reply:
Implemented it. So now the logic is:
- If the song selection is not limited, then the default song stays as is
- If the song selection _is_ limited, and the default song is part of it, then it stays as is.
- If the song selection _is_ limited, and the default song is _not_ part of it, then it'll change to the first song in the limited song list.

It also now displays a note about the currently selected default song, as well as a note that it may not match the select box if the default song is not in the list of limited songs.